test/e2e/failure_domain_lifecycle_test.go

```
@@ -0,0 +1,434 @@
+	defer func() {
+		t.Log("Cleaning up test PVC")
+		if err := kubeClient.CoreV1().PersistentVolumeClaims(ns).Delete(ctx, pvcName, metav1.DeleteOptions{}); err != nil {
+			t.Logf("Failed to delete test PVC %s/%s: %v", ns, pvcName, err)
+		}
+	}()
+
+	// Wait for PVC to bind
+	err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 2*time.Minute, false, func(pollCtx context.Context) (bool, error) {
+		p, err := kubeClient.CoreV1().PersistentVolumeClaims(ns).Get(pollCtx, pvcName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 230) but it returns nil

(nilerr)